worker/peergrouper/desired_test.go

```diff
@@ -172,71 +248,110 @@ func desiredPeerGroupTests(ipVersion TestIPVersion) []desiredPeerGroupTest {
 				id:        "13",
 				wantsVote: true,
 			}),
-			statuses:      mkStatuses("1s 2p 3p", ipVersion),
+			statuses:      mkStatuses("1s 2p 3s", ipVersion),
 			members:       mkMembers("1v 2v 3v", ipVersion),
 			expectVoting:  []bool{true, true, true},
 			expectMembers: nil,
+		}, {
+			about: "two voting members removes vote from secondary (first member)",
+			machines:      mkMachines("11v 12v", ipVersion),
+			members:       mkMembers("1v 2v", ipVersion),
+			statuses:      mkStatuses("1s 2p", ipVersion),
+			expectVoting:  []bool{false, true},
+			expectMembers: mkMembers("1 2v", ipVersion),
+		}, {
+			about:         "two voting members removes vote from secondary (second member)",
+			machines:      mkMachines("11v 12v", ipVersion),
+			members:       mkMembers("1v 2v", ipVersion),
+			statuses:      mkStatuses("1p 2s", ipVersion),
+			expectVoting:  []bool{true, false},
+			expectMembers: mkMembers("1v 2", ipVersion),
+		}, {
+			about:         "three voting members one ready to loose voting -> no consensus",
+			machines:      mkMachines("11v 12v 13", ipVersion),
+			members:       mkMembers("1v 2v 3v", ipVersion),
+			statuses:      mkStatuses("1p 2s 3s", ipVersion),
+			expectVoting:  []bool{true, false, false},
+			expectMembers: mkMembers("1v 2 3", ipVersion),
+		}, {
+			about:         "three voting members remove one, to only one voting member left",
+			machines:      mkMachines("11v 12", ipVersion),
+			members:       mkMembers("1v 2v 3", ipVersion),
+			statuses:      mkStatuses("1p 2s 3s", ipVersion),
+			expectVoting:  []bool{true, false},
+			expectMembers: mkMembers("1v 2", ipVersion),
+		}, {
+			about:         "three voting members remove all, keep primary",
+			machines:      mkMachines("11 12 13", ipVersion),
+			members:       mkMembers("1v 2v 3v", ipVersion),
+			statuses:      mkStatuses("1s 2s 3p", ipVersion),
+			expectVoting:  []bool{false, false, true},
+			expectMembers: mkMembers("1 2 3v", ipVersion),
 		},
 	}
 }
 
-func (s *desiredPeerGroupSuite) TestDesiredPeerGroup(c *gc.C) {
-	DoTestForIPv4AndIPv6(c, s, func(ipVersion TestIPVersion) {
-		for ti, test := range desiredPeerGroupTests(ipVersion) {
-			c.Logf("\ntest %d: %s", ti, test.about)
-			trackerMap := make(map[string]*machineTracker)
-			for _, m := range test.machines {
-				c.Assert(trackerMap[m.Id()], gc.IsNil)
-				trackerMap[m.Id()] = m
-			}
+func (s *desiredPeerGroupSuite) TestDesiredPeerGroupIPV4(c *gc.C) {
+	s.doTestDesiredPeerGroup(c, testIPv4)
+}
```

Comment:
we need to add the matching "TestDesiredPeerGroupIPv6":
func (s *desiredPeerGroupSuite) TestDesiredPeerGroupIPv6(c *gc.C) {
	s.doTestDesiredPeerGroup(c, testIPv6)
}